app/src/main/java/co/ostorlab/insecure_app/bugs/calls/InsecureTemplateInjection.java

```diff
@@ -0,0 +1,67 @@
+package co.ostorlab.insecure_app.bugs.calls;
+import co.ostorlab.insecure_app.BugRule;
+
+import android.content.Context;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import com.github.mustachejava.DefaultMustacheFactory;
+import com.github.mustachejava.Mustache;
+import com.github.mustachejava.MustacheFactory;
+import android.content.res.Resources;
+import java.io.StringWriter;
+
+
+public final class InsecureTemplateInjection extends BugRule {
+
+    private static final String TAG = "RULE";
+    private Context appContext;
+
+    public InsecureTemplateInjection(Context context){
+        appContext = context.getApplicationContext();
+    }
+
+    @Override
+    public String getDescription() {
+        return "insecure template injection.";
+    }
+
+    @Override
+    public void run(String user_input) throws Exception {
+
+        TemplateRenderer templateRenderer = new TemplateRenderer();
+
+        String templateFileName = "template.mustache";
+        String renderedTemplate = templateRenderer.renderTemplate(appContext.getApplicationContext(), templateFileName, user_input);
```

Comment:
This is not vulnerable, you need to concatenate to trigger detection.

Reply:
Regardless `mustache` is logicless meaning it's not vulnerable to template injection by design.